addons/payment/models/payment_method.py

```diff
@@ -289,8 +293,35 @@ def _get_compatible_payment_methods(
                 reason=REPORT_REASONS_MAPPING['express_checkout_not_supported'],
             )
 
+        # Filter pms that does not support manual capture.
+        providers = self.env['payment.provider'].browse(provider_ids)
+        supported_methods = self._get_supported_methods(payment_methods, providers)
+        for method in supported_methods:
+            payment_utils.add_to_report(
+                report,
+                supported_methods,
+                available=supported_methods[method][1],
+                reason=REPORT_REASONS_MAPPING['manual_capture_not_supported'],
+                filter_provider=True,
+            )
+
         return payment_methods
 
+    def _get_supported_methods(self, payment_methods, provider_ids):
+        record = {}
+        for method in payment_methods:
+            available = True
+            supported_providers = []
+            for provider in provider_ids:
+                if provider in method.provider_ids:
+                    if provider.capture_manually and not method.support_manual_capture:
+                        supported_providers.append((provider, False))
+                        available = False
+                    else:
+                        supported_providers.append((provider, True))
+            record[method] = (supported_providers, available)
+        return record
```

Comment:
This seems complex :detective: 
Can we just do something simpler to achieve this :thinking: as we have store information on provider and method :thinking: